src/sfdxProject.ts

```diff
@@ -78,6 +120,57 @@ export class SfdxProjectJson extends ConfigFile<ConfigFile.Options> {
     Object.assign(defaultOptions, options || {});
     return defaultOptions;
   }
+
+  /**
+   * Validates sfdx-project.json against the schema.
+   *
+   * Set the `SFDX_SCHEMA_VALIDATE` environment variable to `true` to throw an error when schema validation fails.
+   * A warning is logged by default.
+   *
+   * ***See*** [sfdx-project.schema.json] (https://raw.githubusercontent.com/forcedotcom/schemas/master/schemas/sfdx-project.schema.json)
+   */
+  public async schemaValidate(): Promise<void> {
+    if (!this.hasRead) {
+      // read calls back into this method after necessarily setting this.hasRead=true
+      await this.read();
+    } else {
+      try {
+        const projectJsonSchemaPath = require.resolve('@salesforce/schemas/sfdx-project.schema.json');
+        const validator = new SchemaValidator(this.logger, projectJsonSchemaPath);
+        await validator.load();
+        await validator.validate(this.getContents());
+      } catch (err) {
+        if (env.getBoolean('SFDX_SCHEMA_VALIDATE', false)) {
```

Comment:
What are your thoughts about putting this in the schemaValidator class? The envar seems generic and i would expect it on all validation.

Reply:
We can't put it there though because that could break existing users of the schema validator.  I'll change the name of the env var to be more specific to sfdx-project.json validation.  Maybe `SFDX_PROJECT_JSON_VALIDATION`